core/src/main/scala/org/broadinstitute/dsde/rawls/spendreporting/SpendReportingService.scala

```diff
@@ -23,377 +22,271 @@ import scala.jdk.CollectionConverters._
+      .groupBy(row => TerraSpendCategories.categorize(row.get("service").getStringValue))
+      .map { case (category, categoryRows) =>
+        SpendReportingForDateRange(
+          sum(categoryRows, "cost"),
+          sum(categoryRows, "credits"),
+          currency.getCurrencyCode,
+          category = Option(category),
+          subAggregation = subKey.map(key => aggregate(categoryRows, SpendReportingAggregationKeyWithSub(key)))
+        )
       }
+      .toList
 
-  def extractSpendReportingResults(rows: List[FieldValueList],
-                                   startTime: DateTime,
-                                   endTime: DateTime,
-                                   workspaceProjectsToNames: Map[GoogleProject, WorkspaceName],
-                                   aggregationKeys: Set[SpendReportingAggregationKeyWithSub]
-  ): SpendReportingResults = {
-    val currency = getCurrency(rows)
-    val spendAggregations = aggregationKeys.map { aggregationKey =>
-      extractSpendAggregation(rows, currency, aggregationKey, workspaceProjectsToNames)
+    def aggregate(rows: List[FieldValueList], key: SpendReportingAggregationKeyWithSub): SpendReportingAggregation = {
+      val spend = key match {
+        case SpendReportingAggregationKeyWithSub(SpendReportingAggregationKeys.Category, sub)  => byCategory(rows, sub)
+        case SpendReportingAggregationKeyWithSub(SpendReportingAggregationKeys.Workspace, sub) => byWorkspace(rows, sub)
+        case SpendReportingAggregationKeyWithSub(SpendReportingAggregationKeys.Daily, sub)     => byDate(rows, sub)
+      }
+      SpendReportingAggregation(key.key, spend)
     }
-    val spendSummary = extractSpendSummary(rows, currency, startTime, endTime)
 
-    SpendReportingResults(spendAggregations.toList, spendSummary)
+    val summary = SpendReportingForDateRange(
+      sum(allRows, "cost"),
+      sum(allRows, "credits"),
+      currency.getCurrencyCode,
+      Option(start),
+      Option(end)
+    )
+    SpendReportingResults(aggregations.map(aggregate(allRows, _)).toList, summary)
   }
 
+}
+
+class SpendReportingService(
+  ctx: RawlsRequestContext,
+  dataSource: SlickDataSource,
+  bigQueryService: cats.effect.Resource[IO, GoogleBigQueryService[IO]],
+  samDAO: SamDAO,
+  spendReportingServiceConfig: SpendReportingServiceConfig
+)(implicit val executionContext: ExecutionContext)
+    extends LazyLogging
+    with RawlsInstrumented {
+
   /**
-    * Ensure that BigQuery results only include one type of currency and return that currency.
+    * Base name for all metrics. This will be prepended to all generated metric names.
+    * Example: dev.firecloud.rawls
     */
-  private def getCurrency(rows: List[FieldValueList]): Currency = {
-    val currencies = rows.map(_.get("currency").getStringValue)
+  override val workbenchMetricBaseName: String = spendReportingServiceConfig.workbenchMetricBaseName
```

Comment:
Just curious, are we doing anything with this at the moment or is this pure wireup for a subsequent PR?